kusto/ingest/internal/resources/resources.go

```diff
@@ -152,6 +154,7 @@ func New(client mgmter) (*Manager, error) {
 	}
 
 	m.kustoTokenCacheExpiration = time.Now().UTC()
+	m.lastFetchTime.Store(time.Now().UTC())
```

Comment:
I'd drop this line.
Consider what will happen if the initial fetch will fail

Reply:
I removed the fetch in the ctor entirely.
It's weird that it does an action that can take seconds when constructing an ingestor object..

Now the background loop will immediately start one, and if for some reason it doesn't it will fall back to synchronous (since we have a condition if lastFetchTime is not set)